### PR TITLE
Initial preferred language shared pref support.

### PR DESCRIPTION
**Pull request by PaulRashidi**

### CODE REVIEW

**Comment by PaulRashidi** [comment]
This is partially operational. You can click the Russian item in the Language select and then navigate elsewhere in the app and the Russian translations apply.

Fine to merge if you're OK with the progress.

File: app/src/main/java/com/g11x/checklistapp/NavigationActivity.java [hunk 1]
```diff
@@ -66,6 +67,15 @@
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
+    overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
+    PreferredLanguageSupport.applyPreferredLanguage(this);
+    final Activity thisActivity = NavigationActivity.this;
```
Comment (aaronj1335) [inline app/src/main/java/com/g11x/checklistapp/NavigationActivity.java:72]: Is this an aesthetic thing? Or is there a technical reason for saving a final reference here? (this is the first time I've seen this pattern)

File: app/src/main/java/com/g11x/checklistapp/language/Language.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+package com.g11x.checklistapp.language;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Custom content languages supported by the app.
+ */
+public enum Language {
+  Russian("ru", "RussianTBU"),
```
Comment (aaronj1335) [inline app/src/main/java/com/g11x/checklistapp/language/Language.java:26]: TBU? To Be Updated?

**Comment by aaronj1335** [comment]
Looks good. I'll go ahead and merge so it's easier to use in our time zone. I had a couple informational questions, looking forward to learning more Java 😸

File: app/src/main/java/com/g11x/checklistapp/NavigationActivity.java (hunk 1 above)
Comment (PaulRashidi) [inline app/src/main/java/com/g11x/checklistapp/NavigationActivity.java:72]: I'm transitioning the variable reference from the local scope to effectively member scope in an anon class so it needs to be final in Java. Many times you'll see parameters in methods declared final for this reason as well.

File: app/src/main/java/com/g11x/checklistapp/language/Language.java (hunk 2 above)
Comment (PaulRashidi) [inline app/src/main/java/com/g11x/checklistapp/language/Language.java:26]: Yes, Hoping @askutarenko would get this.

**Pull request closed without merging**